js/source/tile.js

```diff
@@ -129,11 +129,11 @@ class Tile {
     }
 
     redoPlacement(source) {
-        if (source.type !== 'vector' && source.type !== 'geojson') {
+        if ((source.type !== 'vector' && source.type !== 'geojson') ||
+            !this.hasData() || !this.collisionTile) {
```

Comment:
@jfirebaugh I thought about that, but the problem is that this logic doesn't work currently anyway, because `redoWhenDone` is only processed in the `done` callback which can only happen after a successful `redoPlacement` request (after the tile got `loaded`). Maybe we should rewrite the code to take this case into account in a follow-up PR.